src/Common/Core/Impl/Threading/IMainThread.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Common.Core.Threading {
+    public interface IMainThread {
```

Comment:
Console app, yes - it has a window (the thread that runs the message pump). It depends what you are going to call. You do can mark code as STA or even have to have to do it if your app consumes some COM object marked as STA. 

---
In reply to: [81241676](https://github.com/Microsoft/RTVS/pull/2408#discussion_r81241676) [](ancestors = 81241676,81241433,81223138,81221364)